pkg/controller/servicebindingrequest/servicebinder.go

```diff
@@ -371,6 +372,35 @@ func (b *serviceBinder) setApplicationObjects(
 	sbrStatus.Applications = boundApps
 }
 
+// set default value for application selector
+func initApplicationSelector(applicationSelector *v1alpha1.ApplicationSelector) {
```

Comment:
What happened to `SetDefaults()` function we had i this PR? These should be renamed back to 

```
func (applicationSelector *v1alpha1.ApplicationSelector) SetDefaults() 
```

and moved back to api package.

Reply:
I made this change based on an earlier comment from @isutton 
These kind of functions need not be part of public API.

Reply:
This problem can be solved in two different ways without introducing `SetDefaults` as public API:

* Create a function close to where it is actually needed, since the value is created when it is decoded client-go and not through a constructore; and
* Modify `ApplicationSelector`'s JSON unmarshaller to perform the logic currently in `initApplicationSelector` to guarantee the value coming from the wire has valid data for the program to work with (see https://golang.org/pkg/encoding/json/#Unmarshaler).

Another option is the combination of both: using `initApplicationSelector` in `UnmarshallJSON` in `ApplicationSelector`.

Reply:
Change to `ensureDefaults`

Reply:
Done. ac55e1f3fbd4ac82bae0c560693487412e0f2d50